pkgs/development/tools/clpm/default.nix

```
@@ -0,0 +1,48 @@
+{ stdenv
+, fetchgit
+, wrapLisp
+, sbcl
+, openssl
+}:
+
+stdenv.mkDerivation rec {
+  pname = "clpm";
+  version = "0.3.5";
+
+  src = fetchgit {
+    rev = "v${version}";
+    url = "https://gitlab.common-lisp.net/clpm/clpm";
+    sha256 = "0jivnnp3z148yf4c2nzzr5whz76w5kjhsb97z2vs5maiwf79y2if";
+    fetchSubmodules = true;
```

```suggestion
    url = "https://gitlab.common-lisp.net/clpm/clpm";
    rev = "v${version}";
    fetchSubmodules = true;
    sha256 = "0jivnnp3z148yf4c2nzzr5whz76w5kjhsb97z2vs5maiwf79y2if";
```

Also I would prefer if we used fetchFromGitLab and not venodr 20 submodules.

---

Thanks for the feedback @SuperSandro2000 ! How would you suggest avoid vendoring submodules? Is it possible to run `git submodules init` and `git submodules update` in the configurePhase? It seems that theres no fetchSubmodules in fetchFromGitlab.

---

There isn't? That should be added.

You don't have network access in the all the phases so that wont work.

---

I can use `fetchFromGitHub` and then set `githubBase = "gitlab.common-lisp.net"`, but don't know if that's any better than `fethgit`? It seems like `fetchFromGitLab` could be rewritten to be pretty similar to it's [GitHub equivalent](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/fetchgithub/default.nix). I can look into rewriting `fetchFromGitLab`, need some projects to make me more comfortable with nix. Should we just use `fetchgit` for now?